client/my-sites/marketplace/pages/marketplace-domain-upsell/index.tsx

```diff
@@ -121,7 +129,7 @@ function CalypsoWrappedMarketplaceDomainUpsell(): JSX.Element {
 	};
 
 	const freeWpcomStagingDomain: DomainSuggestions.DomainSuggestion = {
-		domain_name: `${ getSiteNameFromURL( selectedSite?.slug ) }.wpcomstaging.com`,
+		domain_name: `${ getSiteNameFromURL( selectedSite?.slug ?? '' ) }.wpcomstaging.com`,
```

Comment:
I know most of these cases are harmless, because we've had to build robustness into the system already, but in general it might be nicer to avoid calling functions with known-bad data rather than rely on those functions handling bad inputs. in this case we're expecting `selectedSite` to be unavailable? there's no meaningful `domain_name` is there?

```ts
domain_name: selectedSite?.slug ? `${ getSiteNameFromURL( selectedSite.slug ) }.wpcomstaging.com` : '.wpcomstaging.com'
```

the full static strings for the failure cases could also aid in discoverability for later devs who come looking for it, though in this case I realize it's duplicated